### PR TITLE
In-Browser Visual Design

### PR DESCRIPTION
**Pull request by ricardobaeta**
@bzz @smacker @dpordomingo @carlosms 

This is the initial work on Code Annotation Visual Design. Only the annotation UI is designed yet. The other UI's are still subject to further design.

I would like to propose to apply a dark theme to the code visualisation and syntax highlighting as well.

Excuse me for the less mistakes, and some other changes that will imply changing code structure.

I have ideas on further developments that I will address as features requests.  

image

### CODE REVIEW

**Comment by dpordomingo** [comment]
@ricardobaeta I added a screenshot to the PR description. It's something that should be mandatory in PR modifying the UI ;)

File: src/components/Experiment/Footer.less
Comment (dpordomingo) [inline src/components/Experiment/Footer.less]: If you jump to an assignation with already provided feedback, the selector is broken.
image
I think that with should be as wider as the longest value possible.

File: src/pages/Experiment.less
Comment (dpordomingo) [inline src/pages/Experiment.less:199]: when the user is in the `finish` stage, the hover effect is not visible
image

File: src/pages/Experiment.less
Comment (dpordomingo) [inline src/pages/Experiment.less:218]: when the user is in the `finish` stage, the hover effect is not visible
image

File: src/components/Experiment/Footer.less
Comment (dpordomingo) [inline src/components/Experiment/Footer.less]: I found many repeated colors.
- Do you think would be a good idea to have a `palette.less` stylesheet to define each color, and then reuse its values?
- We have 18 different colors. Do you think that the palette could be compressed more considering the most similar?

Here are the colors I found in our app:
(duplicated color, means it is repeated in different places in our stylesheets)

image

in raw:

```
#ffffff x4
#d8dcdc x2  
rgba(0, 0, 0, 0.2)
#cccccc x2
#a0a2a2 x2
#979a9a
#818a8a x2
#808b8c x3
#444444
#2c3132 x2
#191b1c
#0c0d0e x2
#00f3ff x2
#66ffba x2
#f37b6f x2
#006166
#004c2a
#61312c
```

**Review by dpordomingo — CHANGES_REQUESTED** [review]
Wooooooow!!!
Many thanks for taking care of the UI changes!!!
It looks really great now!

I only found a couple of problems and an extra suggestion (in case you want to consider it)

File: src/components/PageHeader.js
Comment (dpordomingo) [inline src/components/PageHeader.js]: The CI fails because [this piece of code is not properly indented](https://travis-ci.org/src-d/code-annotation/builds/336025677#L511).
It should be:
```jsx
        <Navbar.Brand>
          <h1>
            <a href="/">
              <img
                alt="source{d} code annotation"
                src="/sourced-code-annotation-logo.svg"
              />
              code annotation
            </a>
          </h1>
        </Navbar.Brand>
```

**Review by dpordomingo — CHANGES_REQUESTED** [review]

**Comment by bzz** [comment]
It looks awesome! How would we proceed @ricardobaeta from here ?

Usual flow would be that PR creator would solicit the feedback and incorporate it into this branch.

File: src/components/Experiment/Footer.less
Comment (marnovo) [reply src/components/Experiment/Footer.less]: Agreed 👍

**Comment by smacker** [comment]
I'm thinking about to take some code from here and create new PR with few components like that new "select" component, new buttons, breadcrumbs. They won't be used yet, but we will have high-quality code in master with them. After that, we can update UI to this dark theme and use those new components. What do you think guys? @bzz @ricardobaeta @dpordomingo

**Comment by ricardobaeta** [comment]
@smacker I would say that I would address first the comments & suggestion on this PR. We're trying a workflow here, wireframes > implementation > visual design in-browser, and I would say we need to learn if this is a good fit for us and get the results of the experiment. Today I'll address the comments & suggestions and update the PR. What would you say @bzz ?

**Comment by smacker** [comment]
@ricardobaeta the problem is we can't merge it as it is now. Some work which is done here is going to be common for different pages. That's why they must be moved to components.
If you want to abstract them by yourself - it's great.

**Comment by dpordomingo** [comment]
an alternative could be that @ricardobaeta would rebase this PR over current `master`

**Comment by ricardobaeta** [comment]
@dpordomingo The current master would have already the components?

**Comment by dpordomingo** [comment]
nope, but it would solve the conflicts, and the code after being merged could be the starting point for @smacker to extract components.

File: src/components/Experiment/Footer.less
Comment (ricardobaeta) [reply src/components/Experiment/Footer.less]: Yes @dpordomingo . It will be addressed when dealing with the component.

File: src/components/PageHeader.js
Comment (ricardobaeta) [reply src/components/PageHeader.js]: @dpordomingo Thanks, done!

File: src/pages/Experiment.less
Comment (ricardobaeta) [reply src/pages/Experiment.less:199]: @dpordomingo All stages are going to be visually designed as well. From start to finish. For now we're only addressing the annotation screen.

File: src/pages/Experiment.less
Comment (ricardobaeta) [reply src/pages/Experiment.less:218]: @dpordomingo Same as previous comment. Thank you :)

**Comment by ricardobaeta** [comment]
@dpordomingo @smacker Commited changes and forced pushed.
There are things to improve, as talked with Maxim, that will be addressed when dealing with components after this. I'll be around to double check and improve visually.

File: src/pages/Experiment.less
Comment (dpordomingo) [inline src/pages/Experiment.less]: missing `new line`;
all files should finish with (only one) line break.

**Review by dpordomingo — CHANGES_REQUESTED** [review]
LGTM, but not to Travis CI.
You can check the error clicking in `Details` button, and then (in Travis) scrolling down to spot the errors.
To ensuring there is no errors in the things you are about to commit, you can run before commiting:
```bash
CI=1 make test-frontend;
CI=1 make lint;
```
Both commands should pass properly before committing in order to obtain the green icon by our CI.

**Comment by dpordomingo** [comment]
To avoid blocking this PR with color variables https://github.com/src-d/code-annotation/issues/62 and palette https://github.com/src-d/code-annotation/issues/63, I created two separated issues so this PR could be merged once the rest of the feedback has been addressed :D

**Comment by bzz** [comment]
@ricardobaeta @smacker @dpordomingo @ricardobaeta  - is there anything except CI and https://github.com/src-d/code-annotation/pull/53#discussion_r166397066 that blocks merge it? 

If not, let's address it and merge asap, so it can be 'componentized' in subsequent PRs from there.

**Comment by smacker** [comment]
@bzz the main problem is broken pages. Finish/Review pages will look bad if we merge it. That's why I suggest to make components first and then apply them.
I already started working on it.

**Comment by bzz** [comment]
@smacker I think it's ok to merge faster, basically as soon as feedback from above is addressed, even if new pages would not look not so good, but they can be fixed in subsequent PRs. 

> I already started working on it.

Sorry, I must have missed this. Could you please attach a link to the issue as well?

**Comment by smacker** [comment]
as you wish. But I prefer don't break master much.
Issue: https://github.com/src-d/code-annotation/issues/64

File: src/pages/Experiment.less
Comment (ricardobaeta) [reply src/pages/Experiment.less]: Thanks @dpordomingo ! Will iterate, commit and push.

**Comment by ricardobaeta** [comment]
@bzz @dpordomingo @smacker 


Already addressed the new line at end of file as requested. For now, as I understand, from my side all things are ok.

**Comment by dpordomingo** [comment]
Not yet, sorry @ricardobaeta 
You need to obtain a green icon from travis, that now is:

image

And it should be:

image

To do so, you can follow this hint https://github.com/src-d/code-annotation/pull/53#pullrequestreview-94452518

if you need assistance, ping me when I arrive the office (~12:30), and I'll help you with that :dagger:

**Comment by smacker** [comment]
Reworked PR: https://github.com/src-d/code-annotation/pull/65

**Review by dpordomingo — APPROVED** [review]
LGTM
Great work @ricardobaeta
Could you?
- [ ] open an issue with pending things
- [ ] squash before merging

**Review by bzz — APPROVED** [review]
LGTM!

Awesome work, @ricardobaeta ! 

Merging now, but please, address @dpordomingo comments.

**Comment by dpordomingo** [comment]
@ricardobaeta The only thing to address is:
- [ ] open an issue with pending things (the things we saw as broken: footer, selector...)

**Comment by ricardobaeta** [comment]
@dpordomingo https://github.com/src-d/code-annotation/issues/68 ;)

**Pull request merged**
